hw/ip/i2c/doc/programmers_guide.md

```diff
@@ -101,6 +101,30 @@ All other parameters in registers `TIMING2`, `TIMING3`, `TIMING4` are unchanged
 | TIMING1.T_R     | 0                | 134        | 402            | Atypically high line capacitance             |
 | SCL Period      | 1000             | N/A        | 395            | Forced longer than minimum by long T_R        |
 
+## Writing `n` bytes to a device:
+1. Address the device for writing by writing:
+   - FDATA.START = 1;
+   - FDATA.FBYTE = <7-bit address + write bit>.
+2. Fill the TX_FIFO by writing FDATA.FBYTE `n`-1 times.
+3. Send last byte with the stop bit by:
+    - FDATA.STOP = 1;
+    - FDATA.FBYTE = <last byte>.
+
+## Reading `n` bytes from a device:
+1. Address the device for reading by writing:
```

Comment:
Maybe add a TODO to come back to this after we've fixed the missing behavior?
```suggestion
<!-- TODO: Fix #18917 and remove the FMT_FIFO empty check before adding the full transaction to the FMT_FIFO -->
1. Address the device for reading by writing:
```

Reply:
Done